datasets/tuple_ie/tuple_ie.py

```diff
@@ -0,0 +1,161 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TupleInf Open IE Dataset"""
+
+from __future__ import absolute_import, division, print_function
+
+import os
+
+import datasets
+
+
+_CITATION = """\
+@article{Khot2017AnsweringCQ,
+  title={Answering Complex Questions Using Open Information Extraction},
+  author={Tushar Khot and A. Sabharwal and Peter Clark},
+  journal={ArXiv},
+  year={2017},
+  volume={abs/1704.05572}
+}
+"""
+
+_DESCRIPTION = """\
+The TupleInf Open IE dataset contains Open IE tuples extracted from 263K sentences that were used by the solver \
+in “Answering Complex Questions Using Open Information Extraction” (referred as Tuple KB, T). \
+These sentences were collected from a large Web corpus using training questions from 4th and 8th grade as queries. \
+This dataset contains 156K sentences collected for 4th grade questions and 107K sentences for 8th grade questions. \
+Each sentence is followed by the Open IE v4 tuples using their simple format.
+"""
+
+_HOMEPAGE = "https://allenai.org/data/tuple-ie"
+
+_URL = "https://ai2-datasets.s3-us-west-2.amazonaws.com/tuple-ie/TupleInfKB.zip"
+
+_DOMAIN_FILES = {"4th_grade": "4thGradeOpenIE.txt", "8th_grade": "8thGradeOpenIE.txt"}
+
+
+class TupleIEConfig(datasets.BuilderConfig):
+    """BuilderConfig for TupleIE"""
+
+    def __init__(self, *args, domains=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.domains = domains
+
+
+class TupleIE(datasets.GeneratorBasedBuilder):
+    """TupleInf Open IE Dataset"""
+
+    BUILDER_CONFIGS = [
+        TupleIEConfig(
+            name="all",
+            domains=list(_DOMAIN_FILES.keys()),
+            description="collected using training questions from 4th and 8th grade as queries.",
+        )
+    ] + [
+        TupleIEConfig(
+            name=name, domains=[name], description=f"collected using training questions from {name} as queries."
+        )
+        for name in _DOMAIN_FILES.keys()
+    ]
+
```

Comment:
```suggestion
    BUILDER_CONFIG_CLASS = TupleIEConfig
```
we need that to specify that we're using a custom config here